discord/state.py

```diff
@@ -561,16 +592,17 @@ def parse_message_update(self, data):
         if 'components' in data and self._view_store.is_message_tracked(raw.message_id):
             self._view_store.update_from_message(raw.message_id, data['components'])
 
-    def parse_message_reaction_add(self, data):
+    def parse_message_reaction_add(self, data) -> None:
         emoji = data['emoji']
         emoji_id = utils._get_as_snowflake(emoji, 'id')
         emoji = PartialEmoji.with_state(self, id=emoji_id, animated=emoji.get('animated', False), name=emoji['name'])
         raw = RawReactionActionEvent(data, emoji, 'REACTION_ADD')
 
         member_data = data.get('member')
         if member_data:
-            guild = self._get_guild(raw.guild_id)
-            raw.member = Member(data=member_data, guild=guild, state=self)
+            # the guild won't be None here
+            guild = self._get_guild(raw.guild_id) # type: ignore
+            raw.member = Member(data=member_data, guild=guild, state=self) # type: ignore
```

Comment:
This probably isn't true actually.

Reply:
`guild_id` can be None if there's a member object provided?

Reply:
`guild` can be `None` (e.g. guild intent is off)